### PR TITLE
Add docs generator

### PR DESCRIPTION
**Pull request by rogaldh**
MEDIA-1188

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 dist/
 node_modules/
 test/
+/docs
```
Comment (alexkonst) [inline .eslintignore]: Use `docs/` pattern instead of `/docs`. Same for `examples/`.

File: package.json [hunk 2]
```diff
@@ -6,7 +6,9 @@
   "scripts": {
     "build": "webpack",
     "lint": "eslint .",
-    "test": "mocha"
+    "pretest": "npm run lint",
+    "test": "mocha",
+    "docs": "node_modules/.bin/esdoc"
```
Comment (alexkonst) [inline package.json]: Use `esdoc` instead of `node_modules/.bin/esdoc`.

File: package.json [hunk 3]
```diff
@@ -29,6 +31,8 @@
     "babel-loader": "~7.1.2",
     "babel-preset-env": "~1.6.1",
     "chai": "~4.1.2",
+    "esdoc": "^1.0.4",
```
Comment (alexkonst) [inline package.json]: Use `~` instead of `^`.

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    return {
+      CONNECT: 'connect',
+      RECONNECT: 'reconnect',
+      CLOSE: 'close',
+      OFFLINE: 'offline',
+      ERROR: 'error',
+      MESSAGE: 'message'
+    }
+  }
+  constructor (url, options) {
```
Comment (alexkonst) [inline src/index.js]: Files are not identical (lost `will` option). Check `master` branch.

Comment (rogaldh) [reply src/index.js]: Hmm... Lost on rebase : (

**Comment by ivanovaleksey** [comment]
There should be only one commit, shouldn't there?

**Comment by rogaldh** [comment]
Why so?

**Pull request merged**